language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/diagnostic/DiagnosticsHelper.java

```diff
@@ -78,7 +78,11 @@ public synchronized void compileAndSendDiagnostics(ExtendedLanguageClient client
         }
 
         // Replace old entries with an empty list
-        lastDiagnosticMap.keySet().forEach((key) -> diagnosticMap.computeIfAbsent(key, value -> EMPTY_DIAGNOSTIC_LIST));
+        lastDiagnosticMap.forEach((key, value) -> {
+            if (!diagnosticMap.containsKey(key)) {
+                client.publishDiagnostics(new PublishDiagnosticsParams(key, EMPTY_DIAGNOSTIC_LIST));
+            }
+        });
```

Comment:
Isn't the previous does the same?
`lastDiagnosticMap.keySet().forEach((key) -> diagnosticMap.computeIfAbsent(key, value -> EMPTY_DIAGNOSTIC_LIST));`

Reply:
This had be done since the `computeIfAbsent` API add the computed value to the diagnosticMap and then the `lastDiagnosticMap` grows.